From 694f06d03547be61e8d945495ca66e59e2057878 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 30 Oct 2023 14:26:35 +0100
Subject: [PATCH 0020/1695] dts: stm32: add stm32mp257f-ev1 board support

Add STM32MP257F Evaluation board support. It embeds a STM32MP257FAI SoC,
with 4GB of DDR4, TSN switch (2+1 ports), 4*USB typeA, 1*USB2 typeC,
SNOR OctoSPI, mini PCIe, STPMIC2 for power distribution ...

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d2c7f6a17e8c3903eabc2c6e668841ded81f2cd9 dts: stm32: add stm32mp257f-ev1 board support]
Change-Id: I2a2b60f1396a271ea9fb93b4e227570a8c552f18
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
[etienne: cherry-picked from 2032343cbb8fffdcf06f543f34c7c05a10b109a8]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/dts/stm32mp257f-ev1.dts | 30 +++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 core/arch/arm/dts/stm32mp257f-ev1.dts

diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
new file mode 100644
index 000000000..5857f04c0
--- /dev/null
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
+ */
+
+/dts-v1/;
+
+#include "stm32mp257.dtsi"
+#include "stm32mp25xf.dtsi"
+#include "stm32mp25-pinctrl.dtsi"
+#include "stm32mp25xxai-pinctrl.dtsi"
+
+/ {
+	model = "STMicroelectronics STM32MP257F-EV1 Evaluation Board";
+	compatible = "st,stm32mp257f-ev1", "st,stm32mp257";
+
+	aliases {
+		serial0 = &usart2;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x0 0x80000000 0x1 0x00000000>;
+	};
+};
-- 
2.47.2

